src/test/scripts/functions/lineage/RewriteTest6.dml

```diff
@@ -0,0 +1,56 @@
+#-------------------------------------------------------------
+#
+# Copyright 2019 Graz University of Technology
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#-------------------------------------------------------------
+
+X = read($1);
+
+# calculate and same the trans-mult of X
+Res1 = t(X) %*% X;
+while(FALSE) {}	
+
+# find vector V as the last column and matrix X1 as the rest
+nc = ncol(X);
+V = X[,nc];
+X1 = X[, 1:nc-1];
+while(FALSE) {}	
+
+# use X1 and V to derive trans-mult of X
+r1c1 = t(X1) %*% X1;
+while(FALSE) {}	
+r1c2 = t(X1) %*% V;
+while(FALSE) {}	
+r2c1 = t(V) %*% X1;
+while(FALSE) {}	
+r2c2 = t(V) %*% V;
+while(FALSE) {}	
+r1 = cbind(r1c1, r1c2);
+while(FALSE) {}	
+r2 = cbind(r2c1, r2c2);
+while(FALSE) {}	
+#r1 = cbind((t(X1) %*% X1), (t(X1) %*% V));
+#r2 = cbind((t(V) %*% X1), (t(V) %*% V));
+Res2 = rbind(r1, r2);
+
+# verify if the sums are same
+while(FALSE) {}	
+sum1 = sum(Res1);
+sum2 = sum(Res2);
+print("original matrix sum = " + sum1);
+print("rewritten matrix sum = " + sum2);
+
```

Comment:
Checking that the sums match is good, but please do not just print it as this requires a dev to look over it - you might return the difference of sums and check in the java test if this difference is below a threshold.